include/options/options.php

```diff
@@ -0,0 +1,506 @@
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $key The name of the option to check for.
+	 * @return bool
+	 */
+	public function has( string $key ): bool {
+		return isset( $this->options[ $this->current_blog_id ][ $key ] ) && $this->options[ $this->current_blog_id ][ $key ] instanceof Abstract_Option;
+	}
+
+	/**
+	 * Returns the value of the specified option.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $key The name of the option to retrieve.
+	 * @return mixed
+	 */
+	public function &get( string $key ) {
+		if ( ! $this->has( $key ) ) {
+			return null;
+		}
+
+		/** @phpstan-var Abstract_Option */
+		$option = $this->options[ $this->current_blog_id ][ $key ];
+		$value  = $option->get();
+		return $value;
+	}
+
+	/**
+	 * Assigns a value to the specified option.
+	 * This doesn't allow to set an unknown option.
+	 * When doing multiple `set()`, options must be set in the right order: some options depend on other options' value.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $key   The name of the option to assign the value to.
+	 * @param mixed  $value The value to set.
+	 * @return WP_Error
+	 */
+	public function set( string $key, $value ): WP_Error {
+		if ( ! $this->has( $key ) ) {
+			/* translators: %s is the name of an option. */
+			return new WP_Error( 'pll_unknown_option_key', sprintf( __( 'Unknown option key %s.', 'polylang' ), $this->wrap_in_code( $key ) ) );
+		}
+
+		/** @phpstan-var Abstract_Option */
+		$option = $this->options[ $this->current_blog_id ][ $key ];
+
+		if ( $option->set( $value, $this ) ) {
+			// No blocking errors: the value can be stored.
+			$this->modified[ $this->current_blog_id ] = true;
+		}
+
+		// Return errors.
+		return $option->get_errors();
+	}
+
+	/**
+	 * Resets an option to its default value.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $key The name of the option to reset.
+	 * @return mixed The new value.
+	 */
+	public function reset( string $key ) {
+		if ( ! $this->has( $key ) ) {
+			return null;
+		}
+
+		/** @phpstan-var Abstract_Option */
+		$option = $this->options[ $this->current_blog_id ][ $key ];
+
+		if ( $option->reset() ) {
+			$this->modified[ $this->current_blog_id ] = true;
+		}
+
+		return $option->get();
+	}
+
+	/**
+	 * Wraps data into `<code>` tags.
+	 *
+	 * @since 3.7
+	 * @internal
+	 *
+	 * @param array|string $data The data.
+	 * @return array|string
+	 *
+	 * @phpstan-param array<scalar>|scalar $data
+	 * @phpstan-return ($data is array ? array<non-empty-string> : non-empty-string)
+	 */
+	public function wrap_in_code( $data ) {
+		$wrapper = function ( $scalar ) {
+			return "<code>{$scalar}</code>";
+		};
+
+		if ( is_array( $data ) ) {
+			return array_map( $wrapper, $data );
+		}
+
+		return call_user_func( $wrapper, $data );
+	}
+
+	/**
+	 * Tells if an option exists.
+	 * Required by interface `ArrayAccess`.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $offset The name of the option to check for.
+	 * @return bool
+	 */
+	#[\ReturnTypeWillChange]
```

Comment:
Is it still needed?

Reply:
Yes, we can't type-hint `mixed` until php 8.0.

Reply:
As we saw together, this concerns only the return type, not the parameters type. Removed in [3ff4c23](https://github.com/polylang/polylang/pull/1451/commits/3ff4c236b17e14c03a0e5ed8620ece33e09ee331).